### PR TITLE
fix Issue 18099 - betterC check throw statements error!

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/func.d [hunk 1]
```diff
@@ -104,7 +104,8 @@ public:
     override void visit(TryFinallyStatement s)
     {
         DtorExpStatement des;
-        if (fd.nrvo_can && s.finalbody && (des = s.finalbody.isDtorExpStatement()) !is null && fd.nrvo_var == des.var)
+        if (fd.nrvo_can && s.finalbody && (des = s.finalbody.isDtorExpStatement()) !is null &&
+            fd.nrvo_var == des.var && !global.params.betterC)
```
Comment (ibuclaw) [inline src/dmd/func.d]: Can betterC not be used as a feature gate here?

Comment (JinShil) [reply src/dmd/func.d]: Would `&& global.params.useExceptions` be appropriate?

File: test/compilable/test18099.d [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+struct D
+{
+    static struct V {
+	~this() { }
+    }
+    V get() {
+	V v ;
+	return v ;
+    }
+}
```
Comment (JinShil) [inline test/compilable/test18099.d]: Please fix the formatting of this.

**Review by JinShil — DISMISSED** [review]

File: src/dmd/func.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/func.d]: done

File: test/compilable/test18099.d [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    {
+	~this() { }
+    }
+
+    V get()
+    {
+	V v;
+	return v;
+    }
+}
```
Comment (JinShil) [inline test/compilable/test18099.d]: The formatting still looks weird (at least in the GitHub web interface).  Maybe there are some tab characters there that shouldn't be there.

**Review by JinShil — APPROVED** [review]

File: test/compilable/test18099.d (hunk 3 above)
Comment (WalterBright) [reply test/compilable/test18099.d]: fixed the tabs

**Review by wilzbach — APPROVED** [review]
Should I rebase this to `stable`?

**Comment by JinShil** [comment]
> Should I rebase this to stable?

I recommend yes.

**Comment by wilzbach** [comment]
Done.

**Pull request merged**